modules/axtask/src/task.rs

```diff
@@ -429,8 +563,11 @@ impl Deref for CurrentTask {
 }
 
 extern "C" fn task_entry() -> ! {
-    // release the lock that was implicitly held across the reschedule
-    unsafe { crate::RUN_QUEUE.force_unlock() };
+    unsafe {
+        // Clear the prev task on CPU before running the task entry function.
+        crate::current().clear_prev_task_on_cpu();
```

Comment:
Why clear it? It will not used before we call `set_prev_task_on_cpu` on context switch.

Reply:
A switch to B, and task B is not running before, task B should begin with task_entry (not the last line of switch_to),task B is responsible for clear the on_cpu flag of task A, right?

Reply:
Really?
A task only continutes its running after `switch_to` when it has been running before,
otherwise it should start with `task_entry`, am I wrong?

Reply:
`task_entry` is written to `ctx.rip` and the task will jump here after `swtich_to`.

Reply:
But you call `ret` in `context_switch`, after `ret`, the control flow jumps to `task_entry` directly

Reply:
![image](https://github.com/user-attachments/assets/c70212e2-ca25-43c2-ac7b-e04aa8773587)

Reply:
![image](https://github.com/user-attachments/assets/bf1f6818-206f-49ef-92b1-d54b0c064fae)

Reply:
![image](https://github.com/user-attachments/assets/1ac74a11-558b-4959-ba5e-541d965f3d74)

The "new" task will start with `task_entry`, it will not reach here until it calls `yield_xxx`

Reply:
Okay, you are right